### PR TITLE
For discussion: Desktop: Fix build for Qt6

### PR DESCRIPTION
**Pull request by mikeller**
### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

This seems to be needed to make the build work with Qt 6.5 on MacOS (Qt 6 is needed to build for M1 based Macs).

Not sure if there is a better way to do this, but the implicit constructors for `QVariant` from `char *` and `const char *` have been marked as deleted in Qt 6.5.

### Changes made:

1. replace implicit `QVariant` constructors from `char *` and `const char *` with `QVariant::fromValue<>()`.

### Related issues:

Found when investigating #3868.

### CODE REVIEW

File: qt-models/divelocationmodel.cpp [hunk 1]
```diff
@@ -91,11 +91,11 @@ QVariant LocationInformationModel::getDiveSiteData(const struct dive_site *ds, i
 	case Qt::DisplayRole:
 		switch(column) {
 		case DIVESITE: return QVariant::fromValue<dive_site *>((dive_site *)ds); // Not nice: casting away const
-		case NAME: return ds->name;
+		case NAME: return QVariant::fromValue<const char *>(ds->name);
```
Comment (bstoeger) [inline qt-models/divelocationmodel.cpp]: I would think `QString(ds->name)` is more readable.

Comment (bstoeger) [reply qt-models/divelocationmodel.cpp:94]: But it doesn't work... OMG.

Comment (mikeller) [reply qt-models/divelocationmodel.cpp]: > I would think `QString(ds->name)` is more readable.

I tried it that way first, but Qt 6.5.0 does not seem to like it very much:

```
[ 33%] Building CXX object desktop-widgets/CMakeFiles/subsurface_interface.dir/configuredivecomputerdialog.cpp.o
/Users/mikeller/git/subsurface/qt-models/divelocationmodel.cpp:94:21: error: functional-style cast from 'char *const' to 'QVariant' uses deleted function
                case NAME: return QVariant(ds->name);
                                  ^~~~~~~~~~~~~~~~~
/Users/mikeller/qt/6.5.0/macos/lib/QtCore.framework/Headers/qvariant.h:260:5: note: candidate constructor [with T = char *, $1 = false] has been explicitly deleted
    QVariant(T) = delete;
    ^
/Users/mikeller/qt/6.5.0/macos/lib/QtCore.framework/Headers/qvariant.h:250:24: note: candidate constructor
    QT_ASCII_CAST_WARN QVariant(const char *str) noexcept(false)
                       ^
/Users/mikeller/qt/6.5.0/macos/lib/QtCore.framework/Headers/qvariant.h:489:12: note: candidate constructor has been explicitly deleted
    inline QVariant(void *) = delete;
           ^
/Users/mikeller/qt/6.5.0/macos/lib/QtCore.framework/Headers/qvariant.h:206:5: note: candidate constructor
    QVariant(bool b) noexcept;
    ^
/Users/mikeller/qt/6.5.0/macos/lib/QtCore.framework/Headers/qvariant.h:199:5: note: candidate constructor
    QVariant(const QVariant &other);
    ^
/Users/mikeller/qt/6.5.0/macos/lib/QtCore.framework/Headers/qvariant.h:215:5: note: candidate constructor
    QVariant(const QByteArray &bytearray) noexcept;
    ^
/Users/mikeller/qt/6.5.0/macos/lib/QtCore.framework/Headers/qvariant.h:224:5: note: candidate constructor
    QVariant(const QString &string) noexcept;
    ^
/Users/mikeller/qt/6.5.0/macos/lib/QtCore.framework/Headers/qvariant.h:278:12: note: candidate constructor
    inline QVariant(QVariant &&other) noexcept : d(other.d)
           ^
```

Interestingly, building this branch with Qt 5.15.2 on MacOS is working fine for me - my next step is to investigate the difference between my build coniguration (using Qt installed with the official online installer as per `INSTALL`) and the build configuration used for the MacOS check differs, and why one likes the change and the other one doesn't...

Comment (bstoeger) [reply qt-models/divelocationmodel.cpp]: Note that I meant `QString(ds->name)`, not `QVariant(ds->name)`.

Comment (mikeller) [reply qt-models/divelocationmodel.cpp]: Haha, looks like I fell victim to confirmation bias. Using `QString()` does indeed work, and you are correct that it is a lot more concise about what we _want_ to do.

**Comment by bstoeger** [comment]
LGTM, thanks.

**Pull request merged**